### PR TITLE
Add the Makefile and Pipfile

### PR DESCRIPTION
**Pull request by jbasden**
This commit adds a basic makefile and pipfile.
It also changes the name of the testing folder from tests to test.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	rm -rf $(BUILD_DIR)
+
+init:
+	$(PYTHON) -m pip install pipenv --user
+
+dev-deps: init
+	pipenv sync --dev
+
+compile: dev-deps
+	pipenv run pydocstyle app
```
Comment (ylynn) [inline Makefile]: Change app to metricpublisher.

File: Pipfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
+[[source]]
+url = "https://pypi.python.org/simple"
+verify_ssl = true
+name = "pypi"
+
+[packages]
+"boto3" = "*"
```
Comment (jlhood) [inline Pipfile]: Why does boto3 have quotes around it when none of the others do?

**Review by jlhood — APPROVED** [review]
One minor comment. Otherwise, looks good!

**Review by ylynn — CHANGES_REQUESTED** [review]

**Review by ylynn — APPROVED** [review]
Provide some info about test. Note down the command you ran to test, like "make test".

**Pull request merged**